flink-runtime/src/test/java/org/apache/flink/runtime/rest/handler/legacy/backpressure/BackPressureTrackerTestUtils.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.rest.handler.legacy.backpressure;
+
+import org.apache.flink.api.common.ExecutionConfig;
+import org.apache.flink.api.common.JobID;
+import org.apache.flink.api.common.time.Time;
+import org.apache.flink.configuration.Configuration;
+import org.apache.flink.configuration.JobManagerOptions;
+import org.apache.flink.runtime.akka.AkkaUtils;
+import org.apache.flink.runtime.blob.VoidBlobWriter;
+import org.apache.flink.runtime.executiongraph.ExecutionGraph;
+import org.apache.flink.runtime.executiongraph.ExecutionJobVertex;
+import org.apache.flink.runtime.executiongraph.JobInformation;
+import org.apache.flink.runtime.executiongraph.TestingSlotProvider;
+import org.apache.flink.runtime.executiongraph.failover.RestartAllStrategy;
+import org.apache.flink.runtime.executiongraph.restart.NoRestartStrategy;
+import org.apache.flink.runtime.jobgraph.JobVertex;
+import org.apache.flink.runtime.jobgraph.JobVertexID;
+import org.apache.flink.runtime.testingUtils.TestingUtils;
+import org.apache.flink.runtime.testutils.DirectScheduledExecutorService;
+import org.apache.flink.util.SerializedValue;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.concurrent.CompletableFuture;
+
+/**
+ * Utility methods for {@link BackPressureStatsTrackerImplTest} and {@link BackPressureRequestCoordinatorTest}.
+ */
+public class BackPressureTrackerTestUtils {
+
+	public static ExecutionJobVertex createExecutionJobVertex() throws Exception {
```

Comment:
Actually we could use a simpler constructor for `ExecutionJobVertex`, even it is better to use `ExecutionJobVertexTest#createExecutionJobVertex` directly, because `BackPressureTrackerTestUtils` actually did not create back pressure specific components. 

Although the execution graph is mocked inside `ExecutionJobVertexTest#createExecutionJobVertex`, it is not touched by this PR, so we can keep it that or make it as a real graph if you like (not mandatory).

Reply:
Another constructor of ```ExecutionJobVertex``` is package private and we cannot access it here. ```ExecutionJobVertexTest#createExecutionJobVertex``` is private, even we change it to public, we need method like ```ExecutionGraph#getState()#isGloballyTerminalState()```, which is not mocked.

Reply:
I suggest making `ExecutionJobVertexTest#createExecutionJobVertex` as public, because it was defined as static before for further outside usages if possible, then it makes sense to do that.  Or we could extract it as `ExecutionJobVertexTestUtils` class for using by other places.

Also we could use the real `ExecutionGraph` as you did in this class in `ExecutionJobVertexTest#createExecutionJobVertex` and it would benefit also previous tests if not changed much.